tests/integration/test_download.py

```diff
@@ -46,3 +46,36 @@ def test_download_check_md5():
     else:
         assert False, "Should have raised SynapseMd5MismatchError"
 
+
+def test_resume_partial_download():
+    original_file = utils.make_bogus_data_file(40000)
+    original_md5 = utils.md5_for_file(original_file).hexdigest()
+
+    entity = File(original_file, parent=project['id'])
+    entity = syn.store(entity)
+
+    ## stash the original file for comparison later
+    shutil.move(original_file, original_file+'.original')
+    original_file += '.original'
+    schedule_for_cleanup(original_file)
+
+    temp_dir = tempfile.gettempdir()
+
+    url = '%s/entity/%s/file' % (syn.repoEndpoint, entity.id)
+    path = syn._download(url, destination=temp_dir, file_handle_id=entity.dataFileHandleId, expected_md5=entity.md5)
+
+    ## simulate an imcomplete download by putting the
+    ## complete file back into its temporary location
+    tmp_path = os.path.join(temp_dir, "synapse_download_" + str(entity.dataFileHandleId))
```

Comment:
pull out function to compute the temporary file name and use from test and from client.py